packages/realm_dart/lib/src/app.dart

```diff
@@ -61,17 +133,20 @@ class AppConfiguration {
   /// a more complex networking setup.
   final Client httpClient;
 
+  /// Options for the assorted types of connection timeouts for sync connections opened for this app.
+  final SyncTimeoutOptions syncTimeoutOptions;
+
   /// Instantiates a new [AppConfiguration] with the specified appId.
-  AppConfiguration(
-    this.appId, {
-    Uri? baseUrl,
-    String? baseFilePath,
-    this.defaultRequestTimeout = const Duration(seconds: 60),
-    this.metadataEncryptionKey,
-    this.metadataPersistenceMode = MetadataPersistenceMode.plaintext,
-    this.maxConnectionTimeout = const Duration(minutes: 2),
-    Client? httpClient,
-  })  : baseUrl = baseUrl ?? Uri.parse(realmCore.getDefaultBaseUrl()),
+  AppConfiguration(this.appId,
+      {Uri? baseUrl,
+      String? baseFilePath,
+      this.defaultRequestTimeout = const Duration(seconds: 60),
+      this.metadataEncryptionKey,
+      this.metadataPersistenceMode = MetadataPersistenceMode.plaintext,
+      @Deprecated('Use SyncTimeoutOptions.connectTimeout') this.maxConnectionTimeout = const Duration(minutes: 2),
+      Client? httpClient,
+      this.syncTimeoutOptions = const SyncTimeoutOptions()})
```

Comment:
nitpick for better formatting..
```suggestion
      this.syncTimeoutOptions = const SyncTimeoutOptions()},)
```